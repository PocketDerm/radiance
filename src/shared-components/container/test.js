import React from 'react';
import renderer from 'react-test-renderer';

import Container from './index';

describe('Container UI snapshots', () => {
  test('renders basic container', () => {
    const component = renderer.create(<Container>Container Content</Container>);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders container helper components', () => {
    const component = renderer.create(
      <Container>
        <Container.Section>
          Section 1
        </Container.Section>
        <Container.Image src="https://images-na.ssl-images-amazon.com/images/I/61E0pFPOl7L._SL1500_.jpg" />
        <Container.Section>
          <div>Section 2</div>
        </Container.Section>
        <Container.Divider />
        <div css={{"display": "flex"}}>
          <Container.Section>Section 3</Container.Section>
          <Container.VerticalDivider/>
          <Container.Section>Section 4</Container.Section>
        </div>
      </Container>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders clickable container', () => {
    const component = renderer.create(
      <Container type="clickable">Container Content</Container>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders message type container', () => {
    const component = renderer.create(
      <Container type="message">Container Content</Container>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
