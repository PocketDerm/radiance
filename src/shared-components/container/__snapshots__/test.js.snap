// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container UI snapshots renders basic container 1`] = `
.emotion-0 {
  background-color: #ffffff;
  border: 1px solid #ededf0;
}

<div
  className="emotion-0 emotion-1"
>
  Container Content
</div>
`;

exports[`Container UI snapshots renders clickable container 1`] = `
.emotion-0 {
  background-color: #ffffff;
  border: 1px solid #ededf0;
  box-shadow: 0 1px 3px 0 rgba(45,45,48,0.06);
  cursor: pointer;
  -webkit-transition: box-shadow 200ms;
  transition: box-shadow 200ms;
}

.emotion-0:hover {
  box-shadow: 0 1px 3px 0 rgba(45,45,48,0.15);
}

<div
  className="emotion-0 emotion-1"
  type="clickable"
>
  Container Content
</div>
`;

exports[`Container UI snapshots renders container helper components 1`] = `
.emotion-15 {
  background-color: #ffffff;
  border: 1px solid #ededf0;
}

.emotion-0 {
  padding: 1.5rem;
}

@media (min-width:992px) {
  .emotion-0 {
    padding-left: 2.5rem;
    padding-right: 2.5rem;
  }
}

.emotion-2 {
  width: 100%;
  overflow: hidden;
  object-fit: cover;
}

.emotion-6 {
  margin: 0 1.5rem;
  border-bottom: 1px solid #ededf0;
}

@media (min-width:992px) {
  .emotion-6 {
    margin: 0 2.5rem;
  }
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-10 {
  margin: 1.5rem 0;
  border-left: 1px solid #ededf0;
}

@media (min-width:992px) {
  .emotion-10 {
    margin: 2.5rem0;
  }
}

<div
  className="emotion-15 emotion-16"
>
  <div
    className="emotion-0 emotion-1"
  >
    Section 1
  </div>
  <img
    className="emotion-2 emotion-3"
    src="https://images-na.ssl-images-amazon.com/images/I/61E0pFPOl7L._SL1500_.jpg"
  />
  <div
    className="emotion-0 emotion-1"
  >
    <div>
      Section 2
    </div>
  </div>
  <div
    className="emotion-6 emotion-7"
  />
  <div
    className="emotion-14"
  >
    <div
      className="emotion-0 emotion-1"
    >
      Section 3
    </div>
    <div
      className="emotion-10 emotion-11"
    />
    <div
      className="emotion-0 emotion-1"
    >
      Section 4
    </div>
  </div>
</div>
`;

exports[`Container UI snapshots renders message type container 1`] = `
.emotion-0 {
  background-color: #ffffff;
  border: 1px solid #ededf0;
  border-radius: 16px;
  box-shadow: 0 12px 20px 0 rgba(45,45,48,0.05);
}

<div
  className="emotion-0 emotion-1"
  type="message"
>
  Container Content
</div>
`;
